node/src/components/consensus/highway_core/active_validator.rs

```diff
@@ -83,18 +92,58 @@ impl<C: Context> ActiveValidator<C> {
         secret: C::ValidatorSecret,
         start_time: Timestamp,
         state: &State<C>,
+        unit_hash_file: Option<PathBuf>,
     ) -> (Self, Vec<Effect<C>>) {
+        let own_last_unit = unit_hash_file
+            .as_ref()
+            .map(Self::read_last_unit)
+            .transpose()
+            .map_err(|err| match err.kind() {
+                io::ErrorKind::NotFound => (),
```

Comment:
Should we at least log an information (or a warnings) that file was not found? I understand that now we always assume it's there but if it's not then we should communicate that.

Reply:
> I understand that now we always assume it's there

We don't. The idea is that a lack of this file is not an error condition. If you are a validator in an era that is just starting, the file won't be there and it's expected not to be there (how could it be? you haven't voted in that era yet).